lib/components/src/syntaxhighlighter/syntaxhighlighter.tsx

```diff
@@ -124,21 +124,28 @@ export const SyntaxHighlighter: FunctionComponent<Props> = ({
   className = null,
   ...rest
 }) => {
+  const rawCode = typeof children === 'string' && children.trim();
```

Comment:
trim() should not happen before formatting

Reply:
The previous implementation do the following steps:
1. cast children to string
2. trim
3. formatting

https://github.com/storybookjs/storybook/pull/11276/files#diff-315d7b1ff36cd2df0bfa65e00d60dd9cL153

Does it mean the previous implementation had same issue?

Reply:
I thought the previous implementation called trim on the result AFTER formatting?

Reply:
The previous implementation had so many parenthesis that might be misleading. `formatter((children as string).trim())` same as `formatter(children.trim())`

Reply:
As I've found in formatter's `tests` (https://github.com/storybookjs/storybook/blob/da777b8e4bed32ce5ec0b1b60f661e913d8d2f39/lib/components/src/syntaxhighlighter/formatter.test.js#L29-L38) there is no reason to trim a string before or after formatting.

Reply:
Based on formatter's tests I think there is no reason to call trim in `SyntaxHighlighter` component. @ndelangen what are your thoughts?

Reply:
👍